aiida/cmdline/commands/cmd_user.py

```diff
@@ -71,57 +54,31 @@ def user_list():
     type=types.UserParamType(create=True),
     cls=options.interactive.InteractiveOption
 )
-@click.option(
-    '--first-name',
-    prompt='First name',
-    help='First name of the user.',
-    type=click.STRING,
-    contextual_default=partial(get_user_attribute_default, 'first_name'),
-    cls=options.interactive.InteractiveOption
-)
-@click.option(
-    '--last-name',
-    prompt='Last name',
-    help='Last name of the user.',
-    type=click.STRING,
-    contextual_default=partial(get_user_attribute_default, 'last_name'),
-    cls=options.interactive.InteractiveOption
-)
-@click.option(
-    '--institution',
-    prompt='Institution',
-    help='Institution of the user.',
-    type=click.STRING,
-    contextual_default=partial(get_user_attribute_default, 'institution'),
-    cls=options.interactive.InteractiveOption
-)
+@options_setup.SETUP_USER_FIRST_NAME(contextual_default=lambda ctx: ctx.params['user'].first_name)
+@options_setup.SETUP_USER_LAST_NAME(contextual_default=lambda ctx: ctx.params['user'].last_name)
+@options_setup.SETUP_USER_INSTITUTION(contextual_default=lambda ctx: ctx.params['user'].institution)
 @click.option(
     '--set-default',
     prompt='Set as default?',
     help='Set the user as the default user for the current profile.',
     is_flag=True,
-    cls=options.interactive.InteractiveOption
+    cls=options.interactive.InteractiveOption,
+    contextual_default=lambda ctx: ctx.params['user'].is_default
 )
 @click.pass_context
 @decorators.with_dbenv()
-def user_configure(ctx, user, first_name, last_name, institution, set_default):
+def user_configure(ctx, user, first_name, last_name, institution, set_default):  # pylint: disable=too-many-arguments
     """Configure a new or existing user.
 
     An e-mail address is used as the user name.
```

Comment:
Why aren't you also reusing the `SETUP_USER_EMAIL` or using the `USER_MAIL` from `cmdline.params.options.main`?

Reply:
Because the option is quite different. For starters, the argument _name_ (i.e. the name that ends up in the function signature) used in the setup version is `email`. That makes sense for the setup, but here, the option should actually convert that email into a `User` instance. Having that named `email` is a bit weird, and you cannot change that name in a reusable option.

Also there are quite a few keyword arguments that would have to be changed, making reuse almost not useful anymore.